rpmlint/checks/WorldWritableCheck.py

```diff
@@ -0,0 +1,15 @@
+from rpmlint.checks.FileMetadataCheck import FileMeta, FileMetadataCheck
+
+
+class WorldWritableCheck(FileMetadataCheck):
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.whitelists = self.config.configuration['WorldWritableWhitelist']
+        self.verify_whitelists(['path', 'owner', 'group', 'mode'])
+        self.prefix = 'world-writable'
+
+    def check_binary(self, pkg):
+        files = [FileMeta(pfile) for pfile in pkg.files.values()]
+        # select only block and character devices, ignore symlinks
+        files = {f.path: f for f in files if not f.mode[0] in 'bcs' and f.mode[-1] == 't'}
```

Comment:
This is not quite what we need here. This only considers sticky bit directories. What we need are any world-writable files except device files and symlinks. So something like this:

```
files = {f.path: f for f in files if not f.mode[0] in 'bcl' and f.mode[-2] == 'w'}
```

Also you should update the comment line above.

Reply:
Thank you for the fix.